zcash_client_backend/src/data_api.rs

```diff
@@ -541,6 +541,51 @@ pub struct SpendableNotes<NoteRef> {
     orchard: Vec<ReceivedNote<NoteRef, orchard::note::Note>>,
 }
 
+/// A request for transaction data enhancement, spentness check, or discovery
+/// of spends from a given transparent address within a specific block range.
+#[derive(Clone, Debug)]
+pub enum TransactionDataRequest {
+    /// Transaction enhancement (download of complete raw transaction data) is requested.
+    ///
+    /// The caller should respond to this request by calling [`wallet::decrypt_and_store_transaction`]
+    /// to provide data for the requested transaction to the wallet backend. If no data is
+    /// available for the specified transaction,
+    Enhancement(TxId),
+    /// Information about the chain's view of a transaction is requested.
+    ///
+    /// The caller should respond to this request by calling [`WalletWrite::set_transaction_status`]
+    /// to provide transaction status information to the wallet backend.
+    GetStatus(TxId),
+    /// Information about transactions that receive or spend funds belonging to the specified
+    /// transparent address is requested.
+    ///
+    /// Fully transparent transactions, and transactions that do not contain either shielded inputs
+    /// or shielded outputs belonging to the wallet, may not be discovered by the process of chain
+    /// scanning; as a consequence, the wallet must actively query to find transactions that spend
+    /// such funds. Ideally we'd be able to query by [`OutPoint`] but this is not currently
+    /// functionality that is supported by the light wallet server.
+    #[cfg(feature = "transparent-inputs")]
+    SpendsFromAddress {
+        address: TransparentAddress,
+        block_range_start: BlockHeight,
+        block_range_end: Option<BlockHeight>,
+    },
+}
+
+/// Metadata about the status of a transaction obtained by inspecting the chain state.
+#[derive(Clone, Copy, Debug)]
+pub enum TransactionStatus {
+    /// The requested transaction ID was not recognized as belonging to a transaction
+    /// either in the main chain or in the mempool.
+    TxidNotRecognized,
```

Comment:
If the intention is that the caller calls the `GetTransaction` gRPC, then there is a third state the transaction can be in: recognised as mined, but in a non-main chain block (i.e. `Conflicted`, a subset of which would be "mined, then reorg occurred and it wasn't remined before it expired"). See the earlier comment about the mess we need to clean up wrt `GetTransaction`.

Reply:
> if a transaction is mined on a fork but not on the main chain, it should be considered to be in the mempool somewhere until it expires.

This fails in the case of double-spends (same seed in two wallets, and each wallet creates a transaction that spends the same input, and gets mined in a different fork).